File: GREEDY/2_ONE/one.cpp
#include <bits/stdc++.h>

using namespace std;

int n, k;
int result;

int main(void) {
    cin >> n >> k;
    while (true) {
        // N이 K로 나눠지는 수가 될 때까지 빼기
        int target = (n / k) * k;
        result += n - target;
        n = target;
        // N이 K보다 작아지면 (더 이상 나눌 수 없으면) 탈출
        if (n < k) break;
        // N을 K로 나누기
        n /= k;
        result++;
    }
    // 마지막으로 남은 수에 대하여 1씩 빼기
    result += (n - 1);
    cout << result << '\n';
}

File: GREEDY/2_ONE/One.java
import java.util.*;

public class One {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int k = sc.nextInt();
        int result = 0;

        sc.close();

        while(true) {
            // N이 K로 나뉘어질 수 없을 때까지 1을 뺀다.
            int target = (n / k) * k;
            result += n - target;
            n = target;
            // N이 K보다 작아져 나눌 수 없으면 탈출한다.
            if (n < k) break;
            // K로 나눈다.
            n /= k;
            result++;            
        }
        result += (n - 1);
        System.out.println(result);
    }
}
